src/Microsoft.Health.Dicom.Core/Extensions/DicomDatasetExtensions.cs

```diff
@@ -329,5 +329,35 @@ public static void ValidateQueryTag(this DicomDataset dataset, QueryTag queryTag
                 minimumValidator.Validate(dicomElement);
             }
         }
+
+        /// <summary>
+        /// Gets DicomDatasets that matches a DicomSequence.
+        /// </summary>
+        /// <param name="dataset">The DicomDataset to be traversed.</param>
+        /// <param name="searchSequence">The Dicom sequence modelling the path.</param>
+        /// <returns>Lists of DicomDataset that matches the search sequence.</returns>
+        public static IEnumerable<DicomDataset> SearchSequencePath(this DicomDataset dataset, DicomSequence searchSequence)
+        {
+            EnsureArg.IsNotNull(dataset, nameof(dataset));
+            EnsureArg.IsNotNull(searchSequence, nameof(searchSequence));
+
+            var foundDatasets = new List<DicomDataset>();
+
+            var foundSequence = dataset.GetDicomItem<DicomSequence>(searchSequence.Tag);
+            if (foundSequence == null) return foundDatasets;
+
+            var searchDataset = searchSequence.Items.FirstOrDefault();
+
+            foreach (var childDataset in foundSequence.Items)
+            {
+                var elements = childDataset.Intersect(searchDataset, new DicomTagComparer());
```

Comment:
nit: should there be a singleton property for an instance of `DicomTagComparer`, like `DicomTagComparer.Instance`? Similar to how there is `Comparer<T>.Default`

Reply:
Removed